pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go

```diff
@@ -309,6 +314,7 @@ func (c *defaultControllerInstallationControl) reconcile(controllerInstallation
 
 func (c *defaultControllerInstallationControl) delete(controllerInstallation *gardencorev1alpha1.ControllerInstallation, logger logrus.FieldLogger) error {
```

Comment:
Since you are at it, does it make sense to modify the `Reconcile` method from the control interface and pass a real Context? 
```suggestion
type ControlInterface interface {
	Reconcile(context.Context,*gardencorev1alpha1.ControllerInstallation) error
}

```
The real context can then be passed from `func (c *Controller) Run(ctx context.Context, workers int)` in `pkg/controllermanager/controller/controllerinstallation/controllerinstallation.go` 

```suggestion
func (c *defaultControllerInstallationControl) delete(ctx context.Context, controllerInstallation *gardencorev1alpha1.ControllerInstallation, logger logrus.FieldLogger) error {
```

Reply:
Good idea! This also means to amend the `CreateWorker` function which affects all the other controllers in Gardener. Let's do it in a separate PR.